android/app/src/main/java/com/zulipmobile/AnchorScrollView.java

```diff
@@ -417,13 +417,17 @@ private int getMaxScrollY() {
 
     // Zulip changes
     private boolean isChildVisible(@Nonnull View child) {
+        //container
         int height = getHeight();
         int containerTop = getScrollY();
         int containerBottom = containerTop + height;
+
+        //child
         int viewTop = child.getTop();
         int viewBottom = child.getBottom();
 
-        return (viewTop >= containerTop && viewBottom <= containerBottom);
+        //check if child is visible or not
+        return (viewTop >= containerTop && viewBottom <= containerBottom) || child.getHeight() >= height;
```

Comment:
Does this logic handle a very long message correctly? It seems like messages that are taller than the container would always return `true`, even if they are not displayed on the screen.

Reply:
Ya! That might be a case. Will take care of this in my anchorScrollView rework.
Thanks